File: apps/web/src/lib/status-translations.ts
import { getStatusLabel } from '@/lib/i18n';

const STATUS_PT_BR: Record<string, string> = {
  CREATED: 'Criada',
  DISPATCHED: 'Despachada',
  ACTIVE: 'Em Operação',
  BOARDING: 'Embarque',
  BOARDED: 'Embarcado',
  IN_TRANSIT: 'Em Trânsito',
  COMPLETED: 'Concluída',
  NO_SHOW: 'Não Compareceu',
  CANCELLED: 'Cancelada',
  SCHEDULED: 'Agendada',
  STOPPED: 'Parado',
  MOVING: 'Em Movimento',
  ONLINE: 'Online',
  OFFLINE: 'Offline',
};

export function translateStatus(status?: string | null): string {
  const raw = String(status ?? '').trim();
  if (!raw) return '—';
  const normalized = raw.toUpperCase();
  return STATUS_PT_BR[normalized] ?? getStatusLabel(normalized) ?? raw;
}

File: flutter/lib/core/status_translations.dart
String translateStatusPtBr(String? status) {
  final value = (status ?? '').trim().toUpperCase();
  switch (value) {
    case 'CREATED':
      return 'Criada';
    case 'DISPATCHED':
      return 'Despachada';
    case 'ACTIVE':
      return 'Em Operação';
    case 'BOARDING':
      return 'Embarque';
    case 'BOARDED':
      return 'Embarcado';
    case 'IN_TRANSIT':
    case 'IN_PROGRESS':
      return 'Em Trânsito';
    case 'COMPLETED':
      return 'Concluída';
    case 'NO_SHOW':
      return 'Não Compareceu';
    case 'CANCELLED':
      return 'Cancelada';
    case 'SCHEDULED':
      return 'Agendada';
    case 'STOPPED':
    case 'IDLE':
      return 'Parado';
    case 'MOVING':
      return 'Em Movimento';
    case 'ONLINE':
      return 'Online';
    case 'OFFLINE':
      return 'Offline';
    case 'WAITING':
      return 'Aguardando';
    case 'ARRIVED':
      return 'Chegou';
    default:
      return status?.toString() ?? '—';
  }
}
